gno.land/pkg/integration/gnoland.go

```diff
@@ -0,0 +1,334 @@
+		&c.ChainID,
+		"chainid",
+		"dev",
+		"the ID of the chain",
+	)
+
+	fs.StringVar(
+		&c.RootDir,
+		"root-dir",
+		"testdir",
+		"directory for config and data",
+	)
+
+	fs.StringVar(
+		&c.GenesisRemote,
+		"genesis-remote",
+		"localhost:26657",
+		"replacement for '%%REMOTE%%' in genesis",
+	)
+
+	fs.Int64Var(
+		&c.GenesisMaxVMCycles,
+		"genesis-max-vm-cycles",
+		10_000_000,
+		"set maximum allowed vm cycles per operation. Zero means no limit.",
+	)
+}
+
+func execTestingGnoland(t *testing.T, logger log.Logger, gnoDataDir, gnoRootDir string, args []string) (*node.Node, error) {
+	t.Helper()
+
+	// Setup start config.
+	icfg := &IntegrationConfig{}
+	{
+		fs := flag.NewFlagSet("start", flag.ExitOnError)
+		icfg.RegisterFlags(fs)
+
+		// Override default value for flags.
+		fs.VisitAll(func(f *flag.Flag) {
+			switch f.Name {
+			case "root-dir":
+				f.DefValue = gnoDataDir
+			case "chainid":
+				f.DefValue = "tendermint_test"
+			case "genesis-balances-file":
+				f.DefValue = filepath.Join(gnoRootDir, "gno.land", "genesis", "genesis_balances.txt")
+			case "genesis-txs-file":
+				f.DefValue = filepath.Join(gnoRootDir, "gno.land", "genesis", "genesis_txs.txt")
+			default:
+				return
+			}
+
+			f.Value.Set(f.DefValue)
+		})
+
+		if err := fs.Parse(args); err != nil {
+			return nil, fmt.Errorf("unable to parse flags: %w", err)
+		}
+	}
+
+	// Setup testing config.
+	cfg := config.TestConfig().SetRootDir(gnoDataDir)
+	{
+		cfg.EnsureDirs()
+		cfg.Consensus.CreateEmptyBlocks = true
+		cfg.Consensus.CreateEmptyBlocksInterval = time.Duration(0)
+		cfg.RPC.ListenAddress = "tcp://127.0.0.1:0"
+		cfg.P2P.ListenAddress = "tcp://127.0.0.1:0"
+	}
+
+	// Prepare genesis.
+	if err := setupTestingGenesis(gnoDataDir, cfg, icfg, gnoRootDir); err != nil {
+		return nil, err
+	}
+
+	// Create application and node.
+	return createAppAndNode(cfg, logger, gnoRootDir, icfg)
+}
+
+func setupTestingGenesis(gnoDataDir string, cfg *config.Config, icfg *IntegrationConfig, gnoRootDir string) error {
+	newPrivValKey := cfg.PrivValidatorKeyFile()
+	newPrivValState := cfg.PrivValidatorStateFile()
+	priv := privval.LoadOrGenFilePV(newPrivValKey, newPrivValState)
+
+	genesisFilePath := filepath.Join(gnoDataDir, cfg.Genesis)
+	osm.EnsureDir(filepath.Dir(genesisFilePath), 0o700)
+	if !osm.FileExists(genesisFilePath) {
+		genesisTxs := loadGenesisTxs(icfg.GenesisTxsFile, icfg.ChainID, icfg.GenesisRemote)
+		pvPub := priv.GetPubKey()
+
+		gen := &bft.GenesisDoc{
+			GenesisTime: time.Now(),
+			ChainID:     icfg.ChainID,
+			ConsensusParams: abci.ConsensusParams{
+				Block: &abci.BlockParams{
+					// TODO: update limits.
+					MaxTxBytes:   1000000,  // 1MB,
+					MaxDataBytes: 2000000,  // 2MB,
+					MaxGas:       10000000, // 10M gas
+					TimeIotaMS:   100,      // 100ms
+				},
+			},
+			Validators: []bft.GenesisValidator{
+				{
+					Address: pvPub.Address(),
+					PubKey:  pvPub,
+					Power:   10,
+					Name:    "testvalidator",
+				},
+			},
+		}
```

Comment:
I still don't like this :triumph: but I know you want to fix it with the next PR on txtar, so I'll leave this be